### PR TITLE
[ADD] crm_event: combine leads and event types

### PR DESCRIPTION
**Pull request by yajo**
This module extends the functionality of CRM opportunities (and leads, if
enabled) to support linking them to event types and to allow you to keep track
of leads interested in an upcoming event of some type.

This is useful if you organize your events based on the amount of people
interested in a certain type of event.

@Tecnativa TT27664

# Pictures

Just to showcase the features, so you know what to test on runbot.

Let you specify event type and wanted seats in opportunities:

imagen

Quick event type search on opportunities:

imagen

Quick group by event type:

imagen

On pipeline reporting, you can filter by event-related opportunities, group by event type, and measure by wanted seats:

imagen

The same of the above, but for leads (if they are enabled in the settings).

Smart button in event types form, indicating open events of that type (and seats available) and open opportunities of that type (and wanted seats):

imagen

### CODE REVIEW

**Comment by pedrobaeza** [comment]
Call the module `crm_event` for not limiting to event types. The integration can go further in any time.

**Comment by yajo** [comment]
Renamed.

File: crm_event/models/crm_lead.py [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+    event_type_id = fields.Many2one(
+        comodel_name="event.type",
+        index=True,
+        ondelete="restrict",
+        string="Event type",
+        help=(
+            "If this lead/opportunity is related to a specific event type, "
+            "indicate it here."
+        ),
+    )
```
Comment (sergio-teruel) [inline crm_event/models/crm_lead.py:19]: Maybe woulb be good idea add an ondelete??

**Review by sergio-teruel — APPROVED** [review]
Only little comment

File: crm_event/models/crm_lead.py (hunk 1 above)
Comment (yajo) [reply crm_event/models/crm_lead.py:19]: [It's already there](https://github.com/OCA/event/pull/201/files/816ff9125b7c7018c4069ba735912302b97f71a9#diff-6cdd1fe21981658519522db80332ba2dd1899d579ce9b167c5267da1e214057bR13).

**Review by sergio-teruel — APPROVED** [review]

**Review by chienandalu — APPROVED** [review]
:+1:

File: crm_event/views/event_type_view.xml [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+        <field name="groups_id" eval="[(4, ref('sales_team.group_sale_salesman'))]" />
+        <field name="inherit_id" ref="event.view_event_type_form"></field>
+        <field name="arch" type="xml">
+            <div position="before">
+                <!-- HACK Add .oe_inline to button box to avoid broken UI with
+                    other modules that also need to add the button box, such as
+                    event_sale_reservation -->
+                <div class="oe_button_box oe_inline" name="button_box">
+                    <!-- Use same iconography as in contact form -->
+                    <button colspan="2" class="oe_stat_button" type="action" name="%(event.action_event_view)d" icon="fa-ticket" context="{'default_event_type_id': id, 'search_default_event_type_id': id}">
```
Comment (pedrobaeza) [inline crm_event/views/event_type_view.xml]: Can you switch this to a method for avoiding the problem with `id`?

Comment (yajo) [reply crm_event/views/event_type_view.xml]: Done.

**Comment by yajo** [comment]
Well, this is approved an all comments attended, so...

/ocabot merge nobump

**Comment by yajo** [comment]
Sorry, I just had to add a few features before the merge:

imagen

imagen

**Comment by yajo** [comment]
New and hopefully last features:

imagen
imagen

**Pull request merged**